src/TerminalLoggers.jl

```diff
@@ -7,8 +7,13 @@ using Logging:
 import Logging:
     handle_message, shouldlog, min_enabled_level, catch_exceptions
 
+using ProgressMeter:
+    Progress, update!
+
 export TerminalLogger
 
+const ProgressLevel = LogLevel(-1)
```

Comment:
This is the level that has been used for a while in Juno https://junolab.org/ProgressLogging.jl/dev/#ProgressLogging.progress-Tuple{Any}

Reply:
I've just been thinking more about this and about the best way represent progress.

I think it does make sense to have a default urgency associated with a `@progress` frontend, but I also think there's sense in having progress messages at `Debug` level, or progress messages at `Info` level. So I think progress reporting is not inherently tied to the level.

Progress is currently detected via the `_id` and `progress` key. But perhaps progress "should" be determined by the type of the message?

```julia
@info Progress("msg", fraction)
```

I think you suggested something similar elsewhere?

Reply:
Rather than type of message, why not type of value?
`@info "msg" fraction=Progress(fraction)`

TensorBoardLogger is all abut the type of the value,
since it logs them differently

Reply:
> But perhaps progress "should" be determined by the type of the message?
> 
> ```julia
> @info Progress("msg", fraction)
> ```

This is an interesting approach.  It's nice that all the information for "dispatch" is available as the mandatory positional argument.  We can also avoid awkward empty message for logging progress.

> Rather than type of message, why not type of value?
> `@info "msg" fraction=Progress(fraction)`

That's more or less what I suggested here https://github.com/JunoLab/ProgressLogging.jl/issues/7#issuecomment-544033475

But I think it's important to support the spec that already exists.  Apparently this has been used by Juno users for a while.

For the new progress logging API, let's discuss it in https://github.com/JunoLab/ProgressLogging.jl.  At some point it would be nice to have similar package for documenting table logging specification and also providing some utility functions.

Reply:
> But I think it's important to support the spec that already exists.

Agreed completely.  This PR just sparked the thought, so I wrote it down!

Reply:
Thanks for writing it down! I just quoted your comment in https://github.com/JunoLab/ProgressLogging.jl/issues/7#issuecomment-553249772

Reply:
Cool, looks good.